### PR TITLE
Add function to get icon and color code

### PR DESCRIPTION
**Pull request by uga-rosa**

### CODE REVIEW

File: lua/nvim-web-devicons.lua [hunk 1]
```diff
@@ -1079,11 +1079,32 @@ local function set_icon(user_icons)
...
+
+    if not icon_data and ((opts and opts.default) or global_opts.default) then
+      icon_data = default_icon
+    end
+  end
+
+  if icon_data then
+    return icon_data.icon, icon_data.color
+  end
+end
```
Comment (kyazdani42) [inline lua/nvim-web-devicons.lua:1100]: this function could be way more small:
```lua
local has_default = (opts and opts.default) or global_opts.default
local icon_data = icons[name] or icons[ext] or (has_default and default_icon)
if icon_data then return icon_data.icon, icon_data.color
```

Comment (uga-rosa) [reply lua/nvim-web-devicons.lua:1100]: I followed the existing implementation, and I think get_icon can be improved and implemented in the same way.

Comment (uga-rosa) [reply lua/nvim-web-devicons.lua:1075]: I've fixed both, so check them out.

**Review by kyazdani42 — APPROVED** [review]

**Comment by kyazdani42** [comment]
please rebase so we can merge :)

**Comment by uga-rosa** [comment]
@kyazdani42 Is this correct?
I'm beginner to git operations.

**Comment by kyazdani42** [comment]
you did it right, we can merge this. Although you could've just returned the `icon_data.color` in the get_icon function as third return value.

**Comment by uga-rosa** [comment]
Making changes to an existing function that is already widely used was not something I thought I should do. If you, the author, think that's okay, then do so.

**Comment by uga-rosa** [comment]
Oh, I had deleted the update to the readme 😢  I'll leave it up to you along with the above.

**Comment by kyazdani42** [comment]
could you maybe open another PR to add missing doc ?

**Comment by mehalter** [comment]
@kyazdani42 I noticed that this function doesn't respect the colors set by a colorscheme when they are just using highlight groups. Realized this with Feline.nvim which uses `get_icon_color` for setting the color of the filetype icon in the statusbar. Is this considered a bug that should be fixed upstream or should we just rely on using `get_icon` and parsing the highlight group ourselves?

Relevant PR/Issue: https://github.com/feline-nvim/feline.nvim/pull/264, https://github.com/feline-nvim/feline.nvim/issues/263

**Pull request merged**